Improved: Project - MainActionMenu (OFBIZ-12484)

PierreSmits opened the pull request:
Currently the create buttons for the main objects of the project mgt plugin  are located within the find and other widgets/templates of those objects.
In order to improve the usability of OFBiz (and thus the appeal of it for adopters and users) these create buttons/links/etc. should be in a main action menu visible at all times when a user is working within the component.

modified:
ProjectMenus.xml - added MainActionMenu for users with CREATE permission in the component
CommonScreens.xml - added MainActionMenu as an 'include-menu' ref in various common decorator screens.
additional cleaning.

PierreSmits committed:
Improved: Project - MainActionMenu (OFBIZ-12484)

Currently the create buttons for the main objects of the project mgt plugin  are located within the find and other widgets/templates of those objects.
In order to improve the usability of OFBiz (and thus the appeal of it for adopters and users) these create buttons/links/etc. should be in a main action menu visible at all times when a user is working within the component.

modified:
ProjectMenus.xml - added MainActionMenu for users with CREATE permission in the component
CommonScreens.xml - added MainActionMenu as an 'include-menu' ref in various common decorator screens.
additional cleaning.

JacquesLeRoux commented:
Hi @PierreSmits ,

Why removing comments in projectmgr/widget/CommonScreens.xml ?

PierreSmits commented:
HI @JacquesLeRoux,

I guess you are referring to:
`
                <!-- The two default (global) stylesheets are added to the list
                     of stylesheets to the first and second position -->
`
and
`
                <!-- layoutSettings.headerImageUrl can be used to specify an application specific logo; if not set,
                     then the global layoutSettings.commonHeaderImageUrl (specified in GlobalDecorator) will be used. -->
                <!--<set field="layoutSettings.headerImageUrl" value="/images/ofbiz_logo.png" global="true"/>-->`

in the main-decorator screen in CommonScreens.xml.
The motivations to remove this are:

1.  it does not serve any purpose. If it were for educational purposes having it in 1 location (e.g. the example application) would suffice and in any other location it would be superfluous. 
2. it is not present in template for CommonScreens.xml to initiate a new plugin.
3. it is not present in committed plugins like firstdatapaymentgateway, pricat, msggateway, webpos

JacquesLeRoux committed:
Update CommonScreens.xml

Let in place 2 removed comments in. The 2nd is present in 21 other CommonScreens.xml!

JacquesLeRoux commented:
Pierre about:
> in the main-decorator screen in CommonScreens.xml.
> The motivations to remove this are:
> 
> 1.  it does not serve any purpose. If it were for educational purposes having it in 1 location (e.g. the example application) would suffice and in any other location it would be superfluous. 
> 2. it is not present in template for CommonScreens.xml to initiate a new plugin.
> 3. it is not present in committed plugins like firstdatapaymentgateway, pricat, msggateway, webpos

1. I disagree it's always good to have information in CommonScreens.xml. There is such information in example component. It's not enough.
2. I have just added them
3. Those are marginals application. You might want to add these comment there ;)

PierreSmits commented:
@JacquesLeRoux

It is a pity that you regard several plugins as marginal. Maybe you should have voted against having it in the codebase when the moments occurred. More diligence would have led to better code and usability (remember the IDEAL stuff?).
